gateway/api/serverinfo/serverinfo.go

```diff
@@ -18,14 +18,26 @@ import (
 
 var (
 	isOrgMultiTenant = os.Getenv("ORG_MULTI_TENANT") == "true"
-	vinfo            = version.Get()
-	serverInfoData   = openapi.ServerInfo{
+	hasRedactCreds   = func() bool {
+		dlpProvider := os.Getenv("DLP_PROVIDER")
+		switch dlpProvider {
+		case "presidio":
+			return isEnvSet("MSPRESIDIO_ANALYZER_URL") && isEnvSet("MSPRESIDIO_ANONYMIZER_URL")
+		case "gcp":
+			return isEnvSet("GOOGLE_APPLICATION_CREDENTIALS_JSON")
+		default:
+			return false
+		}
+	}()
```

Comment:
The `DLP_PROVIDER` env just make a selection between which provider to use, where gcp is the default option. Therefore, you shouldn't rely that this env is always going to be set. Returning `false` assumes the data masking isn't set, which is a wrong assumption.

You could have data masking working by having this option as empty and `GOOGLE_APPLICATION_CREDENTIALS_JSON` set.

The right implementation based on your logic:
- Check if `MSPRESIDIO_` envs are set and DLP_PROVIDER is `mspresidio`
- Check if `GOOGLE_APPLICATION_CREDENTIALS_JSON` env is set